sdk/go/pulumi-language-go/main.go

```diff
@@ -77,7 +82,17 @@ func findProgram(binary string) (*exec.Cmd, error) {
 		return nil, errors.Errorf("Failed to find go files for 'go run' matching %s", goFileSearchPattern)
 	}
 
-	return exec.Command(program, "run", cwd), nil
+	binaryToRun := path.Join(cwd, pulumiBinaryName)
```

Comment:
Write to temp, not cwd. Use https://pkg.go.dev/io/ioutil#TempFile to decide the full file path.

Reply:
SGTM. fixes a lot of limitations.